enterprise/utils.py

```diff
@@ -734,3 +738,21 @@ def decrypt_string(string, iv):   # pylint: disable=invalid-name
     cipher = Cipher(AES(key), CFB(iv), backend=default_backend())
     decryptor = cipher.decryptor()
     return decryptor.update(string) + decryptor.finalize()
+
+
+def get_or_create_enterprise_learner(enterprise_customer, user):
+    """
+    Retrieve or create the enterprise customer learner for the given parameters.
+    :param enterprise_customer:
+    :param user:
+    :return: enterprise_customer_user instance and created state
+    """
+    enterprise_customer_user, created = EnterpriseCustomerUser.objects.get_or_create(
+        enterprise_customer=enterprise_customer,
+        user_id=user.id
+    )
+    if created:
+        # Announce registration
+        REGISTER_ENTERPRISE_USER.send(sender=None, user=user)
```

Comment:
Another approach is to use the `post_save` signal/receiver instead of using the custom one as doing here.